lib/Sema/CSApply.cpp

```diff
@@ -181,6 +181,21 @@ Solution::resolveConcreteDeclRef(ValueDecl *decl,
   auto sig = decl->getInnermostDeclContext()->getGenericSignatureOfContext();
   auto subst = computeSubstitutions(sig, locator);
 
+  // Lazily instantiate function definitions for class template specializations.
+  // Members of a class template specialization will be instantiated here (not
+  // when imported). If this method has already be instantiated, then this is a
+  // no-op.
+  if (decl->getClangDecl() && isa<clang::CXXMethodDecl>(decl->getClangDecl())) {
```

Comment:
I think that it might be nice to just use a variable with casting here:

```
if (auto *MD = dyn_cast<clang::CXXMethodDecl>(decl->getClangDecl()))
```

The `dyn_cast`  will simply fail and avoids the duplication of the access.  Furthermore, you follow up the check with a forced `cast` to get the method decl below.

Reply:
The problem is I need to make sure `getClangDecl` isn't null, and we can't have both an init-statement and condition before C++17. I just realized I can use `dyn_cast_or_null`, though, so I'll do that.